### ISSUE TITLE
Build error of isir/openface with Release, x64 setting

### ISSUE THREAD

**Issue by sagatake** [isir/greta#40]
Hi,

Since I couldn`t find the issue tab for isir/openface, I put this issue here.

I could successfully build the OpenFaceOfflineZeroMQ with (Release, Win32) or (Debug, x64).
However, I faced following error when I tried to build with (Release, x64).

#############################
System.Windows.Markup.XamlParseException: 'The invocation of the constructor on type 'OpenFaceOffline.MainWindow' that matches the specified binding constraints threw an exception.'
Inner exception
FileNotFoundException: Could not load file or assembly 'CppInerop.dll' or one of its dependencies. The specified module could not be found.
#############################

Even though I disabled the optimized build function, I couldn`t eliminate the error.
How can I build with Release, x64 setting for better frame rate.

Environment:
- Visual Studio 2017
- Windows10
- Intel Core i7-8650, 8 cores
- Intel UHD Graphics 620

Best,
sagatake

**Comment by PhilGauthier** [comment]
Hi,
Did the built of the project UtilLibs/CppInerop in x64 failed ?
Because it build the required CppInerop.dll the exception complains about.
Kind regards,
Philippe

**Comment by sagatake** [comment]
Thank you, Philippe,

Following your suggestion, I tried re-build UtilLibs/Cpplnerop in x64 CPU with x64 setting in Release mode.
And I re-build the OpenFaceOfflineZeroMQ again.
However, I got the same message.
Do you have any idea?

I appreciate all your help.

Best,
Takeshi

**Comment by PhilGauthier** [comment]
Hi,
So the CppInerop project doesn't compile ?

I don't have your errors  when running the code analysis on solution.

We probably have something different either in compilation options, here is mine for CppInerop :
`/GS /GL /W3 /Zc:wchar_t /I"./" /I"D:\dev\OpenFace\lib\local\LandmarkDetector\include" /I"D:\dev\OpenFace\lib\local\FaceAnalyser\include" /I"D:\dev\OpenFace\lib\local\GazeAnalyser\include" /I"D:\dev\OpenFace\lib\local\Utilities\include" /I"D:\dev\OpenFace\lib\3rdParty\CameraEnumerator" /I"D:\dev\OpenFace\lib\3rdParty\OpenBLAS\include" /I"D:\dev\OpenFace\lib\3rdParty\OpenCV\include\opencv" /I"D:\dev\OpenFace\lib\3rdParty\OpenCV\include" /I"D:\dev\OpenFace\lib\3rdParty\dlib\include\dlib\.." /Zi /Ox /Fd"x64\Release\vc141.pdb" /Zc:inline /fp:precise /D "NDEBUG" /D "_WINDOWS" /D "_USRDLL" /D "CPPINEROP_EXPORTS" /D "_WINDLL" /D "_UNICODE" /D "UNICODE" /errorReport:prompt /WX- /Zc:forScope /clr /FU"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\mscorlib.dll" /FU"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\PresentationCore.dll" /FU"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\System.Drawing.dll" /FU"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\WindowsBase.dll" /Oi /MD /std:c++17 /FC /Fa"x64\Release\" /EHa /nologo /Fo"x64\Release\" /Ot /Fp"x64\Release\CppInerop.pch" /diagnostics:classic`

Or in the Visual Studio 2017 configuration, I have :
.NET framework 3.5, 4.7.1, c#, c++ basic features, MSBuild, Visual C++2017 redistributables, VC++ 2017 15.9 lastest v141 tools, debuging and profiling tool, SDK Windows10 for desktop.

Best,
Philippe

**Comment by sagatake** [comment]
Thank you Philippe,

I set your Visual Studio 2017 configurations and compilation options.
But I faced the same problem.

Then, I found that the cause in lib/3rdParty/OpenCV/x64/v141/bin.
Strangely, only this directory was empty even the other directories had .dll or .lib files in it.
After re-running the download_libraries.ps1, everything worked properly!!
I think there was some internet connection problem when I first download those libraries, maybe.

Thank you for your kind help, Philippe!!
I appreciate it.

p.s.
I think it would be nice for others if you activate the issue tab in isir/openface page if possible.


Best,
Takeshi
